core/src/main/java/io/jenkins/pluginhealth/scoring/service/ScoreService.java

```diff
@@ -72,21 +68,30 @@ public ScoreStatistics getScoresStatistics() {
         final int numberOfElement = values.length;
 
         return new ScoreStatistics(
-            Math.round((float) Arrays.stream(values).sum() / numberOfElement),
-            values[0],
-            values[numberOfElement - 1],
-            values[(int) (numberOfElement * .25)],
-            values[(int) (numberOfElement * .5)],
-            values[(int) (numberOfElement * .75)]
-        );
+                Math.round((float) Arrays.stream(values).sum() / numberOfElement),
+                values[0],
+                values[numberOfElement - 1],
+                values[(int) (numberOfElement * .25)],
+                values[(int) (numberOfElement * .5)],
+                values[(int) (numberOfElement * .75)]);
     }
 
     @Transactional
     public int deleteOldScores() {
         return repository.deleteOldScoreFromPlugin();
     }
 
-    public record ScoreStatistics(double average, int minimum, int maximum, int firstQuartile, int median,
-                                  int thirdQuartile) {
+    public record ScoreStatistics(
+            double average, int minimum, int maximum, int firstQuartile, int median, int thirdQuartile) {}
+
+    @Transactional(readOnly = true)
+    public Map<Integer, Long> getScoresDistribution() {
+        final Map<Integer, Long> distribution = Arrays.stream(repository.getLatestScoreValueOfEveryPlugin())
+                .boxed()
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        for (int i = 0; i < 100; i++) {
+            distribution.merge(i, distribution.getOrDefault(i, 0L), Long::sum);
+        }
```

Comment:
This is to make sure we have all possible scores in the map, not only the scores of existing plugins.
For example, we might not have any plugin with a score of 10, which would remove `10` from the graph.

Reply:
But I was using the `merge` method incorrectly.